proto/proto/block_producer.proto

```diff
@@ -58,13 +58,45 @@ message BlockProducerStatus {
 // SUBMIT PROVEN TRANSACTION
 // ================================================================================================
 
+// Error codes specific to the SubmitProvenTransaction endpoint
+enum SubmitProvenTransactionErrorCode {
+    TRANSACTION_DESERIALIZATION_FAILED = 0;
+    INPUT_NOTES_ALREADY_CONSUMED = 1;
+    UNAUTHENTICATED_NOTES_NOT_FOUND = 2;
+    OUTPUT_NOTES_ALREADY_EXIST = 3;
+    INCORRECT_ACCOUNT_INITIAL_COMMITMENT = 4;
+    INVALID_TRANSACTION_PROOF = 5;
+    TRANSACTION_EXPIRED = 6;
+    STALE_INPUTS = 7;
+    DUPLICATE_ACCOUNT_ID_PREFIX = 8;
+    INTERNAL_ERROR = 9;
+}
+
 // Represents the result of submitting proven transaction.
 message SubmitProvenTransactionResponse {
-    // The node's current block height.
-    fixed32 block_height = 1;
+    // Success payload containing the transaction submission result
+    message Success {
+        // The node's current block height.
+        fixed32 block_height = 1;
+    }
```

Comment:
I thought the protobuf messages wouldn't change and we'd pass error codes though gRPC headers. Or would that not work?